nexus/src/app/sagas/instance_create.rs

```diff
@@ -705,39 +705,90 @@ async fn sic_allocate_instance_external_ip_undo(
+    let osagactx = sagactx.user_data();
     let disk_params = sagactx.saga_params::<DiskParams>()?;
     let saga_params = disk_params.saga_params;
     let disk_index = disk_params.which;
     let saga_disks = &saga_params.create_params.disks;
+    let opctx =
+        OpContext::for_saga_action(&sagactx, &saga_params.serialized_authn);
 
     if disk_index >= saga_disks.len() {
         return Ok(None);
     }
 
     let disk = &saga_disks[disk_index];
 
+    // TODO-correctness TODO-security It's not correct to re-resolve the
+    // organization and project names now.  See oxidecomputer/omicron#1536.
+    let organization_name: db::model::Name =
+        saga_params.organization_name.clone().into();
+    let project_name: db::model::Name = saga_params.project_name.clone().into();
+
     match disk {
-        params::InstanceDiskAttachment::Create(_create_params) => {
-            return Err(ActionError::action_failed(
-                "Creating disk during instance create unsupported!".to_string(),
-            ));
+        params::InstanceDiskAttachment::Create(create_params) => {
+            osagactx
+                .nexus()
+                .project_create_disk(
+                    &opctx,
+                    &organization_name,
+                    &project_name,
+                    &create_params,
+                )
+                .await
+                .map_err(ActionError::action_failed)?;
         }
 
         _ => {}
-    }
+    };
 
     Ok(None)
 }
 
 /// Undo disks created during instance creation
-// TODO implement
 async fn sic_create_disks_for_instance_undo(
-    _sagactx: NexusActionContext,
+    sagactx: NexusActionContext,
 ) -> Result<(), anyhow::Error> {
+    let osagactx = sagactx.user_data();
+    let disk_params = sagactx.saga_params::<DiskParams>()?;
+    let saga_params = disk_params.saga_params;
+    let disk_index = disk_params.which;
+    let saga_disks = &saga_params.create_params.disks;
+    let opctx =
+        OpContext::for_saga_action(&sagactx, &saga_params.serialized_authn);
+
+    if disk_index >= saga_disks.len() {
+        return Ok(());
+    }
+
+    let disk = &saga_disks[disk_index];
+
+    // TODO-correctness TODO-security It's not correct to re-resolve the
+    // organization and project names now.  See oxidecomputer/omicron#1536.
+    let organization_name: db::model::Name =
+        saga_params.organization_name.clone().into();
+    let project_name: db::model::Name = saga_params.project_name.clone().into();
+
+    match disk {
+        params::InstanceDiskAttachment::Create(disk_create) => {
+            osagactx
+                .nexus()
+                .project_delete_disk(
+                    &opctx,
+                    &organization_name,
+                    &project_name,
+                    &disk_create.identity.name.clone().into(),
+                )
+                .await
+                .map_err(ActionError::action_failed)?;
+        }
+
+        _ => {}
+    };
```

Comment:
I'm uncertain if this is sufficient for the implementation or if, like for the NICs, we should handle the case where the disk has already been deleted.